### PR TITLE
bark: init at unstable-2023-08-22

### PR DESCRIPTION
**Pull request by wlcx**
bark is a sync audio streamer/receiver. think sonos but open source :)

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by emilylange — CHANGES_REQUESTED** [review]
```patch
diff --git a/pkgs/tools/audio/bark/default.nix b/pkgs/tools/audio/bark/default.nix
index 1c21fa267e6f..bc65b9f5542e 100644
--- a/pkgs/tools/audio/bark/default.nix
+++ b/pkgs/tools/audio/bark/default.nix
@@ -1,30 +1,32 @@
 { rustPlatform
 , fetchFromGitHub
 , lib
-, pkgconfig
+, pkg-config
 , alsa-lib
 , speexdsp
 }:
-let
-  inherit (rustPlatform) buildRustPackage;
-in buildRustPackage {
+
+rustPlatform.buildRustPackage {
   pname = "bark";
-  version = "2023-08-24";
+  version = "unstable-2023-08-22";
+
   src = fetchFromGitHub {
     owner = "haileys";
     repo = "bark";
     rev = "2586b9fb58b496f8ef06f516c9cd3aace77521f7";
     hash = "sha256-sGroae6uJhB9UIpFmvt520Zs9k0ir7H8pGkhKJmVWek=";
   };
-  cargoHash = "sha256-r1y2QW/vAXqfdtOAdY/Bao6UQGhHiTVjPB94UgB/CUs=";
+
+  cargoHash = "sha256-OjlVn4fvKPm3UfqhKkv7cDuvK4mcLcQXPNPK+WScrMc=";
+
   buildInputs = [ alsa-lib speexdsp ];
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ pkg-config ];
 
   meta = {
     description = "Live sync audio streaming for local networks";
     homepage = "https://github.com/haileys/bark";
     license = lib.licenses.agpl3Only;
-    maintainers = [ lib.maintainers.samw ];
+    maintainers = with lib.maintainers; [ samw ];
     platforms = lib.platforms.linux;
   };
 }
```

`pkgconfig` is an alias for `pkg-config`, which is why the ofborg eval failed :)

File: pkgs/tools/audio/bark/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ rustPlatform
+, fetchFromGitHub
+, lib
+, pkg-config
+, alsa-lib
+, speexdsp
+}:
+rustPlatform.buildRustPackage {
```
Comment (anthonyroussel) [inline pkgs/tools/audio/bark/default.nix:8]:
```suggestion
{ lib
, rustPlatform
, fetchFromGitHub
, pkg-config
, alsa-lib
, speexdsp
}:

rustPlatform.buildRustPackage {
```

According to coding convention, lib should come first.

> Arguments should be listed in the order they are used, with the exception of lib, which always goes first.

See https://ryantm.github.io/nixpkgs/contributing/coding-conventions/

Comment (wlcx) [reply pkgs/tools/audio/bark/default.nix:8]: Heh. 

So am I changing this or not? :-)

It passes `nixpkgs-fmt` ;)

Comment (emilylange) [reply pkgs/tools/audio/bark/default.nix:8]: Feel free to dismiss this if you want :)

But you will still have to fix the `vendorHash` and update your commit message from
`bark: init at 2023-08-24` to
`bark: init at unstable-2023-08-22`

Comment (wlcx) [reply pkgs/tools/audio/bark/default.nix]: Both done!

Comment (wlcx) [reply pkgs/tools/audio/bark/default.nix]: I don't understand how the cargo hash changed thought...

**Review by emilylange — APPROVED** [review]

**Comment by wlcx** [comment]
Thanks @emilylange!

**Pull request merged**